src/cpp/rtps/messages/MessageReceiver.cpp

```diff
@@ -662,26 +662,33 @@ bool MessageReceiver::proc_Submsg_Data(
         ch.sourceTimestamp = timestamp_;
     }
 
-#if HAVE_SECURITY
-    if (first_reader->getAttributes().security_attributes().is_payload_protected)
-    {
-        if (participant_->security_manager().decode_serialized_payload(ch.serializedPayload, crypto_payload_,
-                first_reader->getGuid(), ch.writerGUID))
-        {
-            ch.serializedPayload.data = crypto_payload_.data;
-            ch.serializedPayload.length = crypto_payload_.length;
-        }
-    }
-#endif  // HAVE_SECURITY
-
     logInfo(RTPS_MSG_IN, IDSTRING "from Writer " << ch.writerGUID << "; possible RTPSReader entities: " <<
             associated_readers_.size());
 
+    bool is_decoded = false;
```

Comment:
Will this throw a warning when built without security?

In the case of security, I would also initialize it to `!participant_->is_secure()`

Reply:
True about the warning. Added `HAVE_SCURITY` guards around the variable declaration.